proposals/0019-graphql.md

```diff
@@ -0,0 +1,130 @@
+# Solution name
+
+* Proposal: [SSWG-0019](0019-graphql.md)
+* Authors: [Jay Herron](https://github.com/NeedleInAJayStack)
+* Review Manager: TBD
+* Status: **Implemented**
+* Implementation: [GraphQL](https://github.com/GraphQLSwift/GraphQL)
+* Forum Threads: [Pitch](https://forums.swift.org/t/pitch-graphql/59554)
+
+## Package Description
+This library is a Swift port of the [JavaScript reference GraphQL implementation](https://github.com/graphql/graphql-js), written on top of Swift NIO. It provides full server-side instrumentation for defining GraphQL schemas, resolving queries, and encoding/decoding GraphQL messages.
+
+|  |  |
+|--|--|
+| **Package Name** | `GraphQL` |
+| **Module Name** | `GraphQL` |
+| **Proposed Maturity Level** | [Incubation](https://github.com/swift-server/sswg/blob/main/process/incubation.md#process-diagram) |
+| **License** | [MIT](https://mit-license.org/) |
+| **Dependencies** | [swift-nio](https://github.com/apple/swift-nio), [swift-collections](https://github.com/apple/swift-collections) |
+
+## Introduction
+
+GraphQL is an increasingly common web API technology that improves client overfetching and reduces network communication between clients and servers.
+
+This library is a Swift port of the [JavaScript reference GraphQL implementation](https://github.com/graphql/graphql-js), written on top of Swift NIO. It provides full server-side instrumentation for defining GraphQL schemas, resolving queries, and encoding/decoding GraphQL messages.
+
+## Motivation
+
+With GraphQL's growing popularity, the server-side Swift community should have a robust backend implementation to reduce the overhead of adding GraphQL APIs to webserver projects. This overhead is significant; the type system and resolution patterns of GraphQL require a 
+
+A single implementation would be ideal to maximize the benefits of increased attention and community engagement, including faster bug reports and fixes, improved performance, and quicker feature adoption.
+
+## Proposed solution
+
+The proposed solution is a Swift port of the JavaScript reference implementation, built on NIO event loops.
+
+Here is an example of a basic `"Hello world"` GraphQL schema:
+
+```swift
+let schema = try GraphQLSchema(
+    query: GraphQLObjectType(                   // Defines the special "query" type
+        name: "Query",
+        fields: [
+            "hello": GraphQLField(              // Users may query 'hello'
+                type: GraphQLString,            // The result is a string type
+                resolve: { _, _, _, _ in
+                    "world"                     // The result of querying 'hello' is "world"
+                }
+            )
+        ]
+    )
+)
+```
+
+This schema can be queried using the `graphql` function:
+
+```swift
+let result = try graphql(
+    schema: schema,
+    request: "{ hello }",
+    eventLoopGroup: eventLoopGroup
+).wait()
+```
```

Comment:
mind if we show an async/await based snippet here? Since it is supported optionally for 5.5+ clients, might as well show it off :)

Reply:
Great idea. I've changed it to async/await.